swan-lake/integration-tutorials/sending-emails-from-a-service.md

```diff
@@ -99,158 +99,108 @@ Follow the instructions given in this section to develop the service.
+            "name": "John Doe",
+            "dob": "1940-03-19",
+            "ssn": "234-23-525",
+            "address": "California",
+            "phone": "8770586755",
+            "email": "johndoe@gmail.com"
+        },
+        "hospital": "grand oak community hospital",
+        "confirmed": false,
+        "appointmentDate": "2023-10-02"
+    }
     ```
 
-3. Define [configurable variables](https://ballerina.io/learn/by-example/#configurability) for the port on which the listener should listen and the URLs of the backend services. Also, define configurable variables for the host, username, and password of the SMTP client.
+    ![Define records](/learn/images/integration-tutorials/sending-emails-from-a-service/define_records.gif)
 
-    > **Note:** Enable two-factor authentication on your Google account, generate an app password, and use the app password in place of your email password. The app password can be generated with [this link](https://myaccount.google.com/apppasswords?pli=1&rapt=AEjHL4Mf5XDD4rE79YJpP5E2NoNwhvXMET_TWyBcQRn-HMzt0PI8BmptpMGRiBVIamW-0ECgVZtXxMRA19bL4Wfnq_hmjBEMqA).
+    Similarly, generate record types for other payloads from the backend service.
 
-    ```ballerina
-    configurable int port = 8290;
-    configurable string hospitalServicesBackend = "http://localhost:9090";
-    configurable string paymentBackend = "http://localhost:9090/healthcare/payments";
-    configurable string host = "smtp.gmail.com";
-    configurable string username = ?;
-    configurable string password = ?;
-    ```
+    We can use [record type inclusion](https://ballerina.io/learn/by-example/type-inclusion-for-records/) for the `ReservationRequest` record type to include the `Patient` record along with the `cardNo` field within the type of the `patient` field.
 
-4. Define two [`http:Client`](https://ballerina.io/learn/by-example/#http-client) clients to send requests to the backend services and one [`email:SmtpClient`](https://ballerina.io/learn/by-example/#email-client) client to send emails.
+    > **Note:**
+    > While it is possible to work with the JSON payload directly, using record types offers several advantages including enhanced type safety, data validation, and better tooling experience (e.g., completion).
 
-    ```ballerina
-    final http:Client hospitalServicesEP = check initializeHttpClient(hospitalServicesBackend);
-    final http:Client paymentEP = check initializeHttpClient(paymentBackend);
-    final email:SmtpClient smtpClient = check initializeEmailClient();
+4. Define the [HTTP service (REST API)](https://ballerina.io/learn/by-example/#rest-service) that has the resource that accepts user requests, retrieves relevant details from the backend service, and responds to the request.
 
-    function initializeHttpClient(string url) returns http:Client|error => new (url);
+    - Open the [Ballerina HTTP API Designer](https://wso2.com/ballerina/vscode/docs/design-the-services/http-api-designer) in VS Code.
```

Comment:
We need to avoid anonymous records. Is the source up to date with the latest changes? Offline, you showed a separate record type `PatientWithCardNo`?

We need to extract the anonymous record and use inclusion there, not in `ReservationRequest`.